lib/connectors/memory.js

```diff
@@ -723,6 +725,62 @@ Memory.prototype.updateAttributes = function updateAttributes(model, id, data, o
   }
 };
 
+Memory.prototype.replace = function replace(model, id, data, options, cb) {
+  var self = this;
+  if (!id) {
+    var err = new Error('You must provide an id when replacing!');
+    if (cb) {
+      return cb(err);
+    } else {
+      throw err;
+    }
+  }
+
+  // Do not modify the data object passed in arguments
+  // data = Object.create(data);
+
+  this.setIdValue(model, data, id);
+  var cachedModels = this.collection(model);
+  var modelData = cachedModels && this.collection(model)[id];
+  if (modelData) {
+    var id = this.getIdValue(model, data);
```

Comment:
No need to create another `id` variable - you already received `id` in function arguments.